src/main/java/edu/ie3/datamodel/io/source/IdCoordinateSource.java

```diff
@@ -5,14 +5,57 @@
 */
 package edu.ie3.datamodel.io.source;
 
+import edu.ie3.datamodel.utils.CoordinateDistance;
+import edu.ie3.datamodel.utils.GridAndGeoUtils;
 import java.util.Collection;
+import java.util.List;
+import java.util.SortedSet;
+import java.util.stream.Collectors;
 import org.locationtech.jts.geom.Point;
 
 public interface IdCoordinateSource extends DataSource {
 
+  /**
+   * Get the matching coordinate for the given ID
+   *
+   * @param id the ID to look up
+   * @return matching coordinate
+   */
   Point getCoordinate(Integer id);
```

Comment:
Why do you use `Integer` instead of `int`? You might consider using primitives instead of boxed parameters.